src/main.rs

```diff
@@ -14,8 +14,9 @@ mod vm;
 use expr::Node;
 
 fn main() {
-    let args = env::args().collect::<Vec<_>>();
-    let input = args.get(1).expect("Not enough arguments, wanted filename");
+    let input = env::args()
+        .nth(1)
+        .expect("Not enough arguments, wanted filename");
```

Comment:
Only get the used value instead of copying all arguments into an vector causing unneeded copies (minor)